ci.md

```diff
@@ -0,0 +1,229 @@
+# Ignition Continuous Integration
+
+When developers push code changes into Ignition, a series of automated tests
+run for various platforms. This process is called Continuous Integration, or CI.
+This document covers all the CI jobs that are available for Ignition and how to
+interpret their results.
+
+This documentation assumes you have gone over the
+[Contributing guide](https://ignitionrobotics.org/docs/all/contributing) and
+therefore are familiar with:
+
+* Opening pull requests
+* Automated tests
+* Test coverage
+* Static code checking
+* Branching (stable release branches and `main`)
+* Building with cmake or colcon
+* Developer Certificate of Origin (DCO) check
+
+It's also helpful to go over our
+[Release guide](https://github.com/ignitionrobotics/docs/blob/master/release.md#type-of-releases)
+to be familiar with:
+
+* Supported platforms
+* Stable, pre-release and nightly releases
+* Binary releases like Debian
+
+## Types of checks
+
+The main type of check that is performed by CI is compiling the code and running
+all automated tests. Some jobs may also run static checkers, add labels, check DCO
+and code coverage.
+
+All checks are visible at the bottom of a pull request, for example:
+
+![PR checks](images/PR_checks.png)
+
+We use two platforms to run CI ([Jenkins](https://www.jenkins.io/) and
+[GitHub Actions](https://docs.github.com/en/actions)),
+each of them running different builds.
+
+Our Jenkins server is located at
+[https://build.osrfoundation.org/](https://build.osrfoundation.org/).
+It runs builds for all supported operating systems. It also runs an
+[ABI](https://stackoverflow.com/a/2456882/6451468)
+checker for all stable branches.
+
+GitHub actions run on GitHub's servers. Currently, it builds Linux for all
+libraries, and macOS for some of them.
+
+Here's a summary of all checks.
+
+Platform | Job name | OS | What does it do? | Where dependencies come from?
+--- | --- | --- | --- | ---
+Jenkins | `<library>-ci-pr_any-ubuntu-auto-amd64` | Ubuntu | Compile and run tests using CMake and Make | Stable binaries for stable branches, nightlies for `main`
+Jenkins | `<library>-ci-pr_any-homebrew-amd64` | macOS | Compile and run tests using CMake and Make | Homebrew binaries
+Jenkins | `<library>-ci-pr_any-windows*-amd64` | Windows | Compile and run tests using [colcon](https://colcon.readthedocs.io/en/released/#) | External dependencies from [vcpkg](https://docs.microsoft.com/en-us/cpp/build/vcpkg?view=msvc-160), Ignition dependencies built from source with colcon
+Jenkins | `<library>-abichecker-any_to_any-ubuntu_auto-amd64` | Ubuntu | Run ABI checker | Stable binaries
+Actions | `Ubuntu CI / Ubuntu * CI` | Ubuntu Bionic and Focal | Compile and run tests using CMake and Make, run code checker and upload coverage results | Stable binaries for stable branches, nightlies for `main`
```

Comment:
This is a nice chart! Should we mention why we are running tests on both GitHub actions `Ubuntu CI` and Jenkins `<library>-ci-pr_any-ubuntu-auto-amd64`? Readers might wonder why Ubuntu tests are being done twice (and to be honest, I've always wondered this myself as well :upside_down_face:)

Reply:
Added in 191a329